app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,22 @@
+class LikesController < ApplicationController
+  def create
+    new_like = current_user.likes.build
+    new_like.post_id = params[:post_id]
+    new_like.user_id = current_user.id
+
+    check_liked = Like.find(current_user.id)
+    if check_liked.nil?
+      respond_to do |format|
+        format.html do
+          if new_like.save
+            flash[:success] = 'You liked this post'
+          else
+            flash[:danger] = 'There was a problem adding your like'
+          end
+        end
+      end
+    end
+    flash[:danger] = 'You already liked this post'
+    redirect_to user_post_path(current_user.id, params[:post_id])
```

Comment:
- [ ] **[OPTIONAL]** The way your method is set up, the flash message on line 19 will always display even if the ones on lines 12 and 14 have been displayed.

  ![clint-flash](https://user-images.githubusercontent.com/36542986/166810845-ddb44fc4-229b-4bb7-b3ce-1c3d4162b5e0.jpg)

  ## SUGGESTION
  To resolve this you need to redirect - to display the message - and `return` your function - to avoid reaching lines 19 and 20. For example:
  ```
  ...
    format.html do
      if new_like.save
        flash[:success] = 'You liked this post'
        redirect_to user_post_path(current_user.id, params[:post_id]) and return    #=> redirect and return
  ...
  ```

Reply:
Thank you so much for you detailed suggestions. I have resolved the issue successfully